src/js/components/form-elements/TableBody.jsx

```diff
@@ -1,17 +1,41 @@
 import React from 'react';
 
 import PropTypes from 'prop-types';
+import { getTranslate } from 'react-localize-redux';
+import { connect } from 'react-redux';
 
 import TableRow from 'components/form-elements/TableRow';
-
+import Spinner from 'components/spinner/Spinner';
+import { translateWithDefaultMessage } from 'utils/Translate';
 
 const TableBody = (props) => {
   const {
-    fieldsConfig, properties, fields, tableRef = () => {},
+    translate,
+    fieldsConfig,
+    properties,
+    fields,
+    tableRef = () => {
+    },
     addRow = (row = {}) => fields.push(row),
   } = props;
   const RowComponent = properties.subfield ? TableRow : fieldsConfig.rowComponent || TableRow;
 
+  const {
+    emptySubstitutions,
+    loadingSubstitutions,
+  } = properties;
+
+  if (loadingSubstitutions) {
+    return (<Spinner />);
+  }
+
+  if (emptySubstitutions && !fields.length) {
```

Comment:
what is that `fields.length` responsible for here? `fields` is probably an object, so I do not understand the usage of that? Isn't checking `emptySubstitutions` enough?

Reply:
Yes, fields is an object, but it has its length as a property. Checking emptySubstitutions isn't enough, because when you are clicking 'add custom substitution' emptySubsitutions is still 0, but fields.length will be > 0, so message with 'There are no substitutions available' will disapear. When it will be only emptySubstitutions there will be displayed this message and added substitution.